src/chains.rs

```diff
@@ -1049,12 +1049,12 @@ fn trim_tries(s: &str) -> String {
 /// 1. .method();
 /// ```
 /// Which all need parenthesis or a space before `.method()`.
-fn should_add_parens(expr: &ast::Expr) -> bool {
+fn should_add_parens(expr: &ast::Expr, context: &RewriteContext<'_>) -> bool {
     match expr.kind {
-        ast::ExprKind::Lit(ref lit) => crate::expr::lit_ends_in_dot(lit),
+        ast::ExprKind::Lit(ref lit) => crate::expr::lit_ends_in_dot(lit, context),
         ast::ExprKind::Closure(ref cl) => match cl.body.kind {
             ast::ExprKind::Range(_, _, ast::RangeLimits::HalfOpen) => true,
-            ast::ExprKind::Lit(ref lit) => crate::expr::lit_ends_in_dot(lit),
+            ast::ExprKind::Lit(ref lit) => crate::expr::lit_ends_in_dot(lit, context),
```

Comment:
Missed this before, but can we also add test cases to the existing files to make sure we're correctly adding parentheses around method calls on floats and closures that contain a single float literal.

Reply:
I don't think this is actually possible to achieve: https://doc.rust-lang.org/error_codes/E0689.html

Reply:
I think we can ignore the closure test cases, but we should add some for method calls on floats. The following

```rust
fn main() {
    1.0 .neg();
    2. .neg();
}
```

formats as:

```rust
fn main() {
    1.0.neg();
    (2.).neg();
}
```

As long as the code parses as valid rust, rustfmt should be able to handle it even if it errors at compile time.

Reply:
> As long as the code parses as valid rust, rustfmt should be able to handle it even if it errors at compile time

Makes sense

> we should add some for method calls on floats

Done